### PR TITLE
Analytics on Admin Panel; Crowdfunding campaign

### PR DESCRIPTION
**Pull request by UladBohdan**
* Content Analytics on Admin Panel
* Publication TIME on Admin Panel
* Molamola buttons (disabled)
* Copyright label

depends on https://github.com/babajka/babajka-backend/pull/76

### CODE REVIEW

File: components/common/ui/IconImage.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
...967 22.6099 11.9506 19.2802 11.9506ZM19.2802 11.1099C22.1319 11.1099 24.3571 8.83517 24.3571 6C24.3571 3.16484 22.1319 0.890111 19.2802 0.890111C16.4286 0.890111 14.2033 3.16484 14.2033 6C14.2033 8.83517 16.4286 11.1099 19.2802 11.1099ZM29 10.9451H34.4396V11.7692H28.1264V0.230771H29V10.9451ZM45.033 11.7692L43.9121 8.83517H38.1264L37.0055 11.7692H36.0824L40.5659 0.230771H41.489L45.956 11.7692H45.033ZM38.456 8.01099H43.5989L41.0275 1.3022L38.456 8.01099ZM59.4176 0.230771V11.7692H56.7802V5.06044L53.7967 9.95605H53.5L50.5165 5.06044V11.7692H47.8791V0.230771H50.5165L53.6484 5.35714L56.7802 0.230771H59.4176ZM67.5879 12C64.2747 12 61.5879 9.3956 61.5879 6C61.5879 2.6044 64.2747 0 67.5879 0C70.9011 0 73.5879 2.6044 73.5879 6C73.5879 9.3956 70.9011 12 67.5879 12ZM67.5879 9.42857C69.467 9.42857 70.9505 8.06044 70.9505 6C70.9505 3.93956 69.467 2.57143 67.5879 2.57143C65.7088 2.57143 64.2253 3.93956 64.2253 6C64.2253 8.06044 65.7088 9.42857 67.5879 9.42857ZM78.3956 9.23077H82.5165V11.7692H75.7582V0.230771H78.3956V9.23077ZM91.6923 11.7692L91.1154 9.95605H86.8297L86.2527 11.7692H83.3681L87.2912 0.230771H90.6538L94.5769 11.7692H91.6923ZM87.6209 7.48352H90.3242L88.9725 3.24725L87.6209 7.48352Z" />
```
Comment (Drapegnik) [inline components/common/ui/IconImage.js]: Did you use https://github.com/gregberge/svgr?

File: components/common/ui/IconImage.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const IconImage = ({ className, name }) => (
+  <>
+    {name === 'molamola' && (
```
Comment (Drapegnik) [inline components/common/ui/IconImage.js]: it is better for the bundle size if 1 icon = 1 component, so replace with `MolamolaIcon`

File: pages/admin/articles.js [hunk 3]
```diff
@@ -65,16 +65,16 @@ const ARTICLE_COLS = [
   {
     id: 'subtitle',
     className: styles['wir-table__font--size--small'],
-    formatter: text => `${text.slice(0, 100)}...`,
+    formatter: (text) => `${text.slice(0, 100)}...`,
```
Comment (Drapegnik) [inline pages/admin/articles.js]: hmm, but I disabled such format: https://github.com/babajka/babajka-frontend/blob/develop/.prettierrc.js#L5

File: utils/customPropTypes.js [hunk 4]
```diff
@@ -85,6 +85,7 @@ export const ArticleModel = {
   tagsByTopic: PropTypes.objectOf(TagsArray).isRequired,
   theme: ThemeType.isRequired,
   color: PropTypes.string.isRequired,
+  metrics: PropTypes.number,
```
Comment (Drapegnik) [inline utils/customPropTypes.js:88]: number?

File: utils/getters.js [hunk 5]
```diff
@@ -76,6 +76,16 @@ export const getLocalizedArticle = (article, lang) => {
       return acc;
     }, getInitTagsByTopic());
 
+  // Accumulating metrics for all content and interface localizations.
+  const totalMetrics = Object.values(article.locales).reduce(
+    (acc, locale) =>
+      acc +
+      (locale.metrics
+        ? Object.values(locale.metrics).reduce((acc2, counter) => acc2 + parseInt(counter, 10), 0)
```
Comment (Drapegnik) [inline utils/getters.js:84]: why `parseInt`?

this code will be executed for each article: on main page & public article, but we use metrics only on admin. could we somehow optimize that?

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: components/common/ui/IconImage.js (hunk 1 above)
Comment (UladBohdan) [reply components/common/ui/IconImage.js]: I did now!

File: pages/admin/articles.js (hunk 3 above)
Comment (UladBohdan) [reply pages/admin/articles.js]: For some reason prettier changed it's mind again and now it's ok🤷‍♂️

File: utils/customPropTypes.js (hunk 4 above)
Comment (UladBohdan) [reply utils/customPropTypes.js:88]: well after I transform a map - it is a single number

File: utils/getters.js (hunk 5 above)
Comment (UladBohdan) [reply utils/getters.js:84]: if we don't login as admin - backend will not return `locale.metrics` and this block will stop execution very fast. I think that is fine

File: components/common/layout/Footer.js [hunk 6]
```diff
@@ -99,7 +99,7 @@ const Footer = () => (
               <ExternalLink href={CROWDFUNDING_CAMPAIGN.options.link}>
                 <Text id="footer.wir-on" />
                 <span>&nbsp;&nbsp;</span>
-                <IconImage name="molamola" color="#1a9582" />
+                <MolamolaIcon color="#1a9582" />
```
Comment (Drapegnik) [inline components/common/layout/Footer.js]: you pass color, but doesn't use it in `MolamolaIcon` component

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: pages/admin/articles.js (hunk 3 above)
Comment (Drapegnik) [reply pages/admin/articles.js]: I think you forget to run `npm i` after rebase, so old prettier version was used

Comment (UladBohdan) [reply pages/admin/articles.js]: that is possible

File: components/common/layout/Footer.js (hunk 6 above)
Comment (UladBohdan) [reply components/common/layout/Footer.js]: Fixed.

**Review by Drapegnik — APPROVED** [review]

**Pull request merged**